app/markets/ActivityTable.tsx

```diff
@@ -6,81 +6,104 @@ import {
   TableHeader,
   TableRow,
 } from '@/app/components/ui/Table';
-import { formatDateTime, shortenAddress } from '@/utils';
-import { Tag } from '@swapr/ui';
+import { getMarketTransactions } from '@/queries/omen';
+import { formatDateTime, gnosisExplorerAddress, shortenAddress } from '@/utils';
+import { Tag, TagColorSchemeProp } from '@swapr/ui';
+import { useQuery } from '@tanstack/react-query';
 
-const activities = [
-  {
-    user: '0x8a05649a4fa186d032376f6d34f6bf40ca0168ae',
-    action: 'Yes',
-    shares: '1.24',
-    date: 1718999911,
-  },
-  {
-    user: '0x1b3bbecdcb76594e1dc5f7550263cd5ea4a6ab6d',
-    action: 'No',
-    shares: '2.3',
-    date: 1718999111,
-  },
-  {
-    user: '0x7532a3ce4d80cc3fb71d69a5329f76f9f375aa27',
-    action: 'Yes',
-    shares: '32.2',
-    date: 1718912111,
-  },
-  {
-    user: '0x2fe3e7422f35ecbb9dca542279fb2ee172faab32',
-    action: 'Yes',
-    shares: 23.1,
-    date: 1718903353,
-  },
-  {
-    user: '0x269e4c1cb7d8586a99baca522db8c2ef72bbf3ac',
-    action: 'No',
-    shares: 12.2,
-    date: 1718901111,
-  },
-  {
-    user: '0xf9d7aeaa523906f59070eee225f16c7893e2b262',
-    action: 'Yes',
-    shares: 0.42,
-    date: 1718801111,
-  },
-];
+export const ActivityTable = ({ address }: { address: string }) => {
+  const { data, isLoading } = useQuery({
+    queryKey: ['getMarketTransactions', address],
+    queryFn: async () =>
+      getMarketTransactions({
+        first: 20,
+        id: address.toLowerCase(),
+      }),
+  });
+
+  const activities = data?.fpmmTransactions ?? [];
+
+  const getTagColorScheme = (option: string): TagColorSchemeProp => {
+    switch (option.toLowerCase()) {
+      case 'buy':
+        return 'success';
+      case 'sell':
+        return 'danger';
+      case 'add':
+        return 'info';
+      default:
+        return 'info';
+    }
+  };
```

Comment:
Things that we can improve:

- Call it `transactionType` instead of `option`;
- Use `TransactionType` type in the option instead of `string`;
- No need for lower casing;
- Use `TransactionType.Buy`, etc. inside each case clause;
- Declare this function outside the component.

Reply:
Yes, good suggestions, thanks.